src/cpp/rtps/transport/TCPTransportInterface.cpp

```diff
@@ -680,7 +697,6 @@ bool TCPTransportInterface::OpenOutputChannel(
                 {
                     // Disconnect the old channel
                     tcp_sender_resource->channel()->disconnect();
-                    tcp_sender_resource->channel()->clear();
```

Comment:
Why has this been removed?

Reply:
Although with this PR channel reuse seems very difficult, this clear() function could lead to a deadblock since it calls join() with a mutex taken that the other thread might need before effectively finishing its tasks.